.mergify.yml

```diff
@@ -8,13 +8,65 @@ pull_request_rules:
       label:
         add:
           - ceph
+
   # if there is a conflict in a backport PR, ping the author to send a proper backport PR
   - name: ping author on conflicts
     conditions:
       - conflict
     actions:
       comment:
         message: This pull request has merge conflicts that must be resolved before it can be merged. @{{author}} please rebase it. https://rook.io/docs/rook/master/development-flow.html#updating-your-fork
+
+  # automerge on master only under certain strict conditions
+  - name: automerge merge master with specific label and approvals with code change
+    conditions:
+      - author=mergify[bot]
+      - base=master
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - label!=do-not-merge
+      - label=ready-to-merge
+      - 'check-success=DCO'
+      - 'check-success=continuous-integration/jenkins/pr-head'
+      - 'check-success=canary'
+      - 'check-success=unittests'
+      - 'check-success=golangci-lint'
+      - 'check-success=codegen'
+      - 'check-success=lint'
+      - 'check-success=modcheck'
+      - 'check-success=pvc'
+      - 'check-success=pvc-db'
+      - 'check-success=pvc-db-wal'
+      - 'check-success=encryption-pvc'
+      - 'check-success=encryption-pvc-db'
+      - 'check-success=encryption-pvc-db-wal'
+      - 'check-success=encryption-pvc-kms-vault-token-auth'
```

Comment:
Every time we add a new action we will need to add it to this list. Is there no way to have it check for overall success instead?

Reply:
Not possible and explicitly unrecommended https://docs.mergify.io/conditions.html#validating-all-status-check